src/App.js

```diff
@@ -1,29 +1,30 @@
-import React, {useCallback, useState} from 'react';
-import {Switch, BrowserRouter, Route, Redirect} from 'react-router-dom'
+import React, { useState } from 'react';
+import { Switch, BrowserRouter, Route } from 'react-router-dom';
 import Header from 'pages/shared/Header/Header';
 import Footer from 'pages/shared/Footer/Footer';
+import NotFound from 'pages/NotFound/NotFound';
 import Home from 'pages/Home/Home';
-import MovieDetails from "pages/MovieDetails/MovieDetails";
+import MovieDetails from 'pages/MovieDetails/MovieDetails';
 
 const App = () => {
+  const [searchIconVisible, setShowSearchIconVisible] = useState(false);
 
-    const [searchIconVisible, setShowSearchIconVisible] = useState(false);
+  const handleSearchIcon = visibility => {
+    setShowSearchIconVisible(visibility);
+  };
 
-    const handleSearchIcon = (visibility) => {
-        setShowSearchIconVisible(visibility);
-    };
-
-    return (
-        <BrowserRouter>
-            <Header searchIconVisible={searchIconVisible} handleSearchIcon={handleSearchIcon}/>
-            <Switch>
-                <Route exact path="/:id" component={MovieDetails}/>
-                <Route exact path="/" render={() => <Home handleSearchIcon={handleSearchIcon}/>}/>
-                <Redirect from="*" to="/" />
-            </Switch>
-            <Footer/>
-        </BrowserRouter>
-    );
-}
+  return (
+    <BrowserRouter>
+      <Header searchIconVisible={searchIconVisible} handleSearchIcon={handleSearchIcon} />
+      <Switch>
+        <Route exact path='/' render={() => <Home handleSearchIcon={handleSearchIcon} />} />
+        <Route exact path='/film/:id' component={MovieDetails} />
+        <Route exact path='/search' component={Home} />
```

Comment:
according to evaluation criteria `/search` should have query param as well. 
F.e.: `/search/like` - wherer 'like' is a word you are looking for